src/test-exhaust.cpp

```diff
@@ -173,13 +174,26 @@ void TestComputeFunctions() {
 
 } // namespace
 
-int main(void) {
+int main(int argc, char** argv) {
+    uint64_t test_complexity = 4;
+    if (argc > 1) test_complexity = std::stoull(std::string(argv[1]), nullptr);
```

Comment:
$ ./test 99999999999999999999
terminate called after throwing an instance of 'std::out_of_range'
  what():  stoull
Aborted (core dumped)

Maybe claml and handle that a bit more gracefully? (I was actually trying to overflow the below << 7 w/ sanitizer but never got it to fail there because its unsigned, though the result will still be unexpected. :) )

Reply:
(and maybe you don't want to check an unsigned value for the less than portion of <= 0 there.

Reply:
reproduced (debian testing)
```
((HEAD detached at origin/pr/33))$ ./test 99999999999999999999
terminate called after throwing an instance of 'std::out_of_range'
  what():  stoull
Aborted
```
idem for negative input -> unsigned wrapping
```
((HEAD detached at origin/pr/33))$ ./test 0
Iteration count must be > 0.jon@purity:~/projects/bitcoin/minisketch ((HEAD detached at origin/pr/33))$ ./test -1
Running libminisketch tests with complexity=18446744073709551615
^C
```
result with invalid argument
```
((HEAD detached at origin/pr/33))$ ./test a
terminate called after throwing an instance of 'std::invalid_argument'
  what():  stoull
Aborted
```

Reply:
Done, made it catch various exceptions, and added range checks.